modules/tinymce/src/themes/silver/main/ts/ui/core/ControlUtils.ts

```diff
@@ -10,9 +10,9 @@ const composeUnbinders = (f: VoidFunction, g: VoidFunction): VoidFunction => ()
   g();
 };
 
-const onSetupEditableToggle = <T extends Toolbar.ToolbarButtonInstanceApi | Menu.MenuItemInstanceApi>(editor: Editor): (api: T) => VoidFunction =>
+const onSetupEditableToggle = <T extends Toolbar.ToolbarButtonInstanceApi | Menu.MenuItemInstanceApi>(editor: Editor, require?: () => boolean): (api: T) => VoidFunction =>
   onSetupEvent<T>(editor, 'NodeChange', (api) => {
-    api.setEnabled(editor.selection.isEditable());
+    api.setEnabled(editor.selection.isEditable() && (Type.isFunction(require) ? require() : true));
```

Comment:
I don't have a good alternative off the top of my head, but I think we need to find a better way than this.

Reply:
Maybe we could make the entire this a predicate that is passed in and defaults to `(editor: editor) => editor.selection.isEditable()` then we wouldn't need the checking if the require is a function and also the toggling could be something custom.
But the risk with that is that someone might forget to add that `editor.selection.isEditable()` call.

Reply:
Yeah, I wasn't sure how to do it better than this. As for passing in the whole predicate makes me concerned of desyncs where the default becomes something else, but some other places are never updated.

Reply:
I have two ideas that would make me happier:
- rename the parameter to `enabledPredicate`, and perhaps made it mandatory (or defaulted to `Fun.always` since it's used 18 times).
- Stop using `onSetupEditableToggle()` for the one case where this extra predicate applies. It's a very simple function, duplicating it seems better than adding complexity.